File: Java/Types of Matrix/Matrix multiplication.java
class matrix_multiplication {
    int[][] matrixMultiply(int[][] A, int[][] B) {
        int[][] result = new int[A.length][B[0].length];
        int calc;
        for (int row = 0; row < A.length; row++) {
            for (int col = 0; col < B[row].length; col++) {
                calc = 0;
                for (int m = 0; m < B.length; m++) {
                    calc += A[row][m] * B[m][col];
                }
                result[row][col] = calc;
            }
        }
        return result;
    }
    public static void main(String[] args) {
        int[][] A = {
            { 1, -1 },
            { 0,  2 }
        };
        int[][] B = {
            { 1, 3, 0 },
            { 2, 0, 1 }
        };

        matrix_multiplication get = new matrix_multiplication();
        if(A[0].length == B.length){
            int[][] sum = get.matrixMultiply(A, B);

            // Print multiplied matrix
            System.out.println("Multiplied matrix: ");
            for (int row = 0; row < sum.length; row++) {
                for (int col = 0; col < sum[row].length; col++) {
                    System.out.print("\t" + sum[row][col]);
                }
                System.out.println();
            }
        }else{
            System.out.println("Error! Multiplication not possible");
        }
    }
}

File: C++/Types of Matrix/Matrix multiplication.cpp
#include <iostream>
#include <vector>
using namespace std;

vector <vector<int>> matrixMultiply(vector <vector<int>> A, vector <vector<int>> B) {
    vector <vector<int>> result (A.size(), vector <int> (B[0].size()));
    int calc;
    for (int row = 0; row < A.size(); row++) {
        for (int col = 0; col < B[row].size(); col++) {
            calc = 0;
            for (int m = 0; m < B.size(); m++) {
                calc += A[row][m] * B[m][col];
            }
            result[row][col] = calc;
        }
    }
    return result;
}
int main(){
    vector <vector<int>> A = {
        { 1, -1 },
        { 0,  2 }
    };
    vector <vector<int>> B = {
        { 1, 3, 0 },
        { 2, 0, 1 }
    };

    if(A[0].size() == B.size()){
        vector <vector<int>> sum = matrixMultiply(A, B);

        // Print multiplied matrix
        cout << "Multiplied matrix: " << endl;
        for (int row = 0; row < sum.size(); row++) {
            for (int col = 0; col < sum[row].size(); col++) {
                cout << "\t" << sum[row][col];
            }
            cout << endl;
        }
    }else{
        cout << "Error! Multiplication not possible" << endl;
    }
}